tests/test_hardware.py

```diff
@@ -0,0 +1,27 @@
+import unittest
+
+from lsst.ts.tunablelaser.hardware import CPU8000
+
+
+class TestCPU8000(unittest.TestCase):
+    def setUp(self):
+        self.cpu8000 = CPU8000(None)
+        self.sim_cpu8000 = CPU8000(None, True)
+
+    def test_publish(self):
+        self.cpu8000.port = unittest.mock.Mock()
+        self.cpu8000.port.perform_magic = unittest.mock.Mock()
+        self.cpu8000.power_register.port = unittest.mock.Mock()
+        self.cpu8000.display_current_register.port = unittest.mock.Mock()
+        self.cpu8000.fault_register.port = unittest.mock.Mock()
+        self.cpu8000.publish()
+
+    def test_set_simulation_mode(self):
+        self.cpu8000.set_simulation_mode(True)
+        self.assertEqual(self.cpu8000.power_register.simulation_mode, True)
+        self.assertEqual(self.cpu8000.display_current_register.simulation_mode, True)
+        self.assertEqual(self.cpu8000.fault_register.simulation_mode, True)
+
+    def test_str(self):
+        self.assertEqual(str(self.cpu8000),
+                         "CPU8000: \n Power: None \n Display Current: None \n Fault code: None \n")
```

Comment:
If this str representation is simply a nice way to show what's going on then I suggest a less detailed test, e.g. a few self.assertIn calls for obvious strings that should be present.

Also please consider:
* Eliminating the trailing whitespace (e.g. "Power: None \n" -> "Power: None\n")
* Using f string formatting in the __str__ methods, for clarity and compactness.
* I will go so far as to suggest *always* using f strings except possibly in log messages when you really want deferred formatting (handy for messages that are verbose and unlikely to be seen, but deferred formatting risks that the message format will turn out to be wrong and the message cannot be output).